salesagility/SuiteCRM
# Fix Activities and History Queries

## Description




Fix aliasing for `date_end`/`date_due` fields so that the columns from each of the three modules (Calls, Meetings and Tasks) match consistently in the UNION ALL queries.



### Changes:
- Calls and Meetings should use the `date_end` field aliased to `date_due` (or whatever as long as it's consistent everywhere which has been the issue thus far).
- Task should use the `date_due` field with an inverse alias of `date_end` oddly.

## Motivation and Context

Inconsistency in aliases across UNION table columns caused queries to intermittently failed under certain circumstances such as having at least one of each type of history record in the subpanel etc.



## How To Test This

- Create any record and relate at least one task, meeting and call in both the Activities and History subpanels. 
- Load the detail view of that record and it should no longer die.
- Check the suite logs and there should no longer be a failing query.



## Types of changes




### Final checklist
